Content/posts/launching-language-and-platform-package-compatibility.md

```diff
@@ -41,32 +43,36 @@ We then decided on a list of Swift compiler versions we’d like to check compat
 * Swift 5.2 (5.2.4)
 * Swift 5.3 (beta)
 
-That’s a lot of builds! That adds up to 40 builds per package, but that’s not all. What if there’s a stable release and a beta release? The stable version might support Swift 4.2 and higher, and the new beta might drop support for anything less than Swift 5.2. Finally, as we also track the status of the default branch, we build for that too. Very quickly, we’ve arrived at a place where we might need to run 120 builds *per package*! With almost 3,200 packages in the index, that’s a *lot* of potential builds - about 380,000! 😅
+That adds up to 40 builds per package, but that’s just the beginning. What if there’s a stable release and a beta release? The stable version might support Swift 4.2 and higher, and the new beta might drop support for anything less than Swift 5.2. Finally, as we also track the status of the default branch, we build for that too. Very quickly, we’ve arrived at a place where we might need to run 120 builds *per package*! With almost 3,200 packages in the index, that’s a *lot* of potential builds - about 380,000! 😅
 
 In practice, it’s less than that as some combinations don’t make sense. For example, there’s no way to build on Apple Silicon with anything less than Swift 5.3, and most packages don’t have a beta release, but it’s still a *lot* of builds. We’ve processed about 200,000 builds as I write this, and we’re still going.
 
 If you’ve been following [these tweets](https://twitter.com/daveverwer/status/1291808885259620353), it should be obvious what all that processing was! Let’s take a look at the last **30 days** of CPU graphs for our production server, a 2018 Mac mini with 32Gb RAM and a 6-core i7 CPU:
 
 ![A graph showing a few spikes of CPU activity, followed by a sustained 100% CPU load.](/images/production-server-thirty-day-cpu-graph.png)
 
-You can see a few of our final test runs in that graph, and then we started the full processing run. Since then, we’ve kept the CPU completely pegged for more than two weeks. We’ve also had our staging Mac mini, and a DTK working on builds too.
+You can see a few of our final test runs in that graph, and then we started processing for real. Since then, we’ve kept the CPU completely pegged for more than two weeks. We’ve also had our staging Mac mini, and a DTK working on builds too.
```

Comment:
and a 2016 MacBook Pro! Not sure if it's worth mentioning but I suspect we wouldn't have managed to keep the Linux queue down without it.

Reply:
Did we use Ceres in production in the end? I didn't know that.

Reply:
Pushed a fix.

Reply:
Yeah, heavily. Gaëlle asked earlier this week - why is this thing making so much noise? 😅

Reply:
We're running four linux builds in parallel, Ceres is running two of those (+1 dev, +1 prod). The main queues were able to keep it up mainly because they didn't have to deal with the Linux pressure.